internal/fwserver/server_readdatasource.go

```diff
@@ -75,17 +77,30 @@ func (s *Server) ReadDataSource(ctx context.Context, req *ReadDataSourceRequest,
 		readReq.ProviderMeta = *req.ProviderMeta
 	}
 
+	readReq.ClientCapabilities = req.ClientCapabilities
```

Comment:
Nit: `ClientCapabilities` could be set in the `readReq` instantiation directly since its not conditional -- that should be the typical pattern in these request/response mapping functions. 👍 

Similar elsewhere.